### ISSUE TITLE
Birthdays not synced from Nextcloud 27 to Android 12/13

### ISSUE THREAD

**Issue by rfc2822** [bitfireAT/vcard4android#24]
### Discussed in https://github.com/bitfireAT/davx5-ose/discussions/385

<div type='discussions-op-text'>

<sup>Originally posted by **fwiep** August 22, 2023</sup>
Hello,

Using DAVx5 v4.3.5.2 from F-Droid on several devices, both Android 12.1 (2x custom built LineageOS) and Android 13 (1x stock Samsung UI). 

Since about one week, all contacts no longer have their birthdays set on the Android device. In the Nextcloud, all is well and all the dates are present and shown on screen.

I have tried to clear the Contacts-app cache, even installed an extra (official Google) Contacts app to compare the result. On all three devices, not a single birthday is shown. The Contact app allows me to add a birthday to the contact, but via DAVx5, no birthday gets synchronized.

When I edit, say the 'Notes' field of a contact in the Nextcloud, then sync through DAVx5, this modification shows up instantly. 

How can I get the birthdays back into my Android contacts?

How should I debug this issue? I have collected a sync's verbose logfile, but no error is shown.

Thanks for your help,
FWieP</div>

**Comment by rfc2822** [comment]
@ArnyminerZ Can you please have a look (reproduce with a test first)? I think `EventBuilder.buildEvent` should handle `date`s with time zone, too (in the `when (date) { … }`) to support this specific case. Since we can't store time zone info in the contacts provider, we have to convert it into UTC.

**Comment by ArnyminerZ** [comment]
Reproduced with a copy-paste of `testType_Birthday`:
```kotlin
@Test
fun testType_BirthdayWithTimezone() {
    EventBuilder(Uri.EMPTY, null, Contact().apply {
        birthDay = Birthday("19510311T000000+0100")
    }, false).build().also { result ->
        assertEquals(CommonDataKinds.Event.TYPE_BIRTHDAY, result[0].values[CommonDataKinds.Event.TYPE])
    }
}
```

**Comment by devvv4ever** [comment]
Does not seem to be fixed in 4.3.9 yet. 

This is the vcard:

```
BEGIN:VCARD
VERSION:3.0
PRODID:-//Sabre//Sabre VObject 4.4.2//EN
UID:LcjQnxjYhKYC9CZtYSgAnkDwrg6QDk-1345-0-0
N:Doe;John;;;
FN:John Doe
BDAY:20010415T000000+0200
ITEM1.ADR:;; ;;;;Nederland
ITEM1.X-ABLABEL:Home
CATEGORIES:_personen
GENDER:M
END:VCARD
```

Please can you check again @ArnyminerZ

**Comment by rfc2822** [comment]
Guess that's because`VCardDateFormat.parse` creates an `OffsetDateTime`, not a `ZonedDateTime`:

https://github.com/mangstadt/ez-vcard/blob/d4c9ab9fb11f4ed6d83c32bdaa2089a73043bd9b/src/main/java/ezvcard/util/VCardDateFormat.java#L185

We should

- [ ] support OffsetDateTime in addition to / instead of? ZonedDateTime,
- [ ] have a test that contains such a BDAY directly as raw string and sees whether it's converted correctly (to an OffsetDateTime).

**Comment by fwiep** [comment]
Hello again,

I had to wait for F-Droid to build and distribute the 4.3.10-ose package. Now I have tested and the problem is not yet fixed - but I think I know why.

If I compare the used library versions in the app's About dialog, I see the same vcard4android version as with 4.3.9-ose (`b376d2e`), and *not* version `a7a1f42` which contains the most recent fix.

Is this an issue with F-Droid's build process, or is there another way to get a true-full-up-to-date apk for testing?

Thanks and kind regards,
FWieP
